core/arch/arm/include/kernel/thread.h

```diff
@@ -497,38 +497,71 @@ void thread_add_mutex(struct mutex *m);
  */
 void thread_rem_mutex(struct mutex *m);
 
+/*
+ * Disables and empties the peralloc RPC cache one reference at a time. If
+ * all threads are idle this function returns true and a cookie of one shm
+ * object which was removed from the cache. When the cache is empty
+ * *is_empty is set to true and the cache is disabled else false. If one
+ * thread isn't idle this function returns false.
+ */
+bool thread_disable_prealloc_rpc_cache(bool *is_empty, uint64_t *cookie);
+
+/*
+ * Enabled the prealloc RPC cache. If all threads are idle the cache is
+ * enabled and this function returns true. If one thread isn't idle this
+ * function return false.
+ */
+bool thread_enable_prealloc_rpc_cache(void);
+
+enum rpc_buf_type {
+	RPC_BUF_TYPE_APPL,
+	RPC_BUF_TYPE_KERNEL,
```

Comment:
Corresponds to `OPTEE_MSG_RPC_SHM_TYPE_APPL` and `OPTEE_MSG_RPC_SHM_TYPE_KERNEL` in `optee_msg.h`. I'll update with a description of the enum.

`RPC_BUF_TYPE_KERNEL` can be used to allocate memory for non-secure translation tables when we add support for that.